shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLMultiStatementsHandler.java

```diff
@@ -0,0 +1,198 @@
+    }
+    
+    private LogicSQL createLogicSQL(final String sql, final SQLStatement sqlStatement) {
+        SQLStatementContext<?> sqlStatementContext = SQLStatementContextFactory.newInstance(
+                metaDataContexts.getMetaDataMap(), Collections.emptyList(), sqlStatement, connectionSession.getSchemaName());
+        return new LogicSQL(sqlStatementContext, sql, Collections.emptyList());
+    }
+    
+    private ExecutionContext createExecutionContext(final LogicSQL logicSQL) {
+        SQLCheckEngine.check(logicSQL.getSqlStatementContext().getSqlStatement(), logicSQL.getParameters(),
+                metaDataContexts.getMetaData(connectionSession.getSchemaName()).getRuleMetaData().getRules(), connectionSession.getSchemaName(), metaDataContexts.getMetaDataMap(), null);
+        return kernelProcessor.generateExecutionContext(logicSQL, metaDataContexts.getMetaData(connectionSession.getSchemaName()), metaDataContexts.getProps());
+    }
+    
+    @Override
+    public ResponseHeader execute() throws SQLException {
+        Collection<ShardingSphereRule> rules = metaDataContexts.getMetaData(connectionSession.getSchemaName()).getRuleMetaData().getRules();
+        DriverExecutionPrepareEngine<JDBCExecutionUnit, Connection> prepareEngine = new DriverExecutionPrepareEngine<>(
+                JDBCDriverType.STATEMENT, metaDataContexts.getProps().<Integer>getValue(ConfigurationPropertyKey.MAX_CONNECTIONS_SIZE_PER_QUERY),
+                (JDBCBackendConnection) connectionSession.getBackendConnection(), (JDBCBackendStatement) connectionSession.getStatementManager(), new StatementOption(false), rules);
+        ExecutionGroupContext<JDBCExecutionUnit> executionGroupContext = prepareEngine.prepare(anyExecutionContext.getRouteContext(), samplingExecutionUnit());
+        for (ExecutionGroup<JDBCExecutionUnit> eachGroup : executionGroupContext.getInputGroups()) {
+            for (JDBCExecutionUnit each : eachGroup.getInputs()) {
+                prepareBatchedStatement(each);
+            }
+        }
+        return executeBatchedStatements(executionGroupContext);
+    }
+    
+    private Collection<ExecutionUnit> samplingExecutionUnit() {
+        Collection<ExecutionUnit> result = new ArrayList<>(dataSourcesToExecutionUnits.size());
+        for (List<ExecutionUnit> each : dataSourcesToExecutionUnits.values()) {
+            result.add(each.get(0));
+        }
+        return result;
+    }
+    
+    private void prepareBatchedStatement(final JDBCExecutionUnit each) throws SQLException {
+        Statement statement = each.getStorageResource();
+        for (ExecutionUnit eachExecutionUnit : dataSourcesToExecutionUnits.get(each.getExecutionUnit().getDataSourceName())) {
+            statement.addBatch(eachExecutionUnit.getSqlUnit().getSql());
+        }
+    }
+    
+    private UpdateResponseHeader executeBatchedStatements(final ExecutionGroupContext<JDBCExecutionUnit> executionGroupContext) throws SQLException {
```

Comment:
directly in the proxy-front module,   compared to the behavior consistency of non-multi statement is "run in the opposite direction "

Reply:
Sorry, I dont understand. Could you please clarify?

Reply:
> Sorry, I dont understand. Could you please clarify?

pass.